CONTRIBUTING.rst

```diff
@@ -236,134 +236,160 @@ You should configure VS code as follow :
+| `pre-commit.ci   <https://pre-commit.ci/>`__                                                  | Run pre-commit (as defined in `.pre-commit-config.yaml` )        |            |  👍   |                                           |
++-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+
+| `CodeBeat      <https://codebeat.co/>`__                                                      | Automated code review and analysis tools                         | -          | 👍    | all versions according to GitHub workflow |
++-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+
+| `CodeScene <https://codescene.com/>`__                                                        | Automated code review and analysis tools                         | -          | 👍    | ?                                         |
++-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+
+| `CodeFactor <https://www.codefactor.io/>`__                                                   | Automated code review and analysis tools                         | -          | 👍    | ?                                         |
++-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+
 
 
 
-.. note::
-	To maintain consistency, make sure to stick to the version defined in the `.pre-commit-config.yaml` file. This version will be used in the CI.
 
 
 
 
+Code testing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 
-**Ruff**
+Every code change must be tested !
+
+
+
+
+**Pytest**
+
+DISDRODB tests are written using the third-party `pytest <https://docs.pytest.org>`_ package.
+
 
-Ruff should be used that way :
+
+The tests located in the `disdrodb/tests` folder are used to test various functions of the code and are automatically run when changes are pushed to the main repository through a GitHub Pull Request.
 
 .. code-block:: bash
 
-	ruff check .
+	pytest disdrodb/tests
 
 
-.. note::
-	To maintain consistency, make sure to stick to the version and the rule configuration defined in the `.pre-commit-config.yaml` file. This information is used in the CI.
 
 
+To create a new reader test, simply add a small, single-station dataset and the associated files (issue, metadata), and expected data, in the following manner:
+
+| 📁 disdrodb
+| ├── 📁 tests
+|     	├── 📁 pytest_files
+|           ├── 📁 check_readers
```

Comment:
I wonder if we don't want to shortcut that to `data/readers/` or create disdrodb/test_data directory to avoid too long fpaths

Reply:
`disdrodb/test/pytest_files/check_readers` vs `disdrodb/test_data/readers`